### PR TITLE
Store TeX source

### LINKED ISSUE #193: Store source TeX in `Image` object, serialize
This is a baby version of #187 and should be rushed so that new files have the data

### PR DESCRIPTION
**Pull request by pihart**
Resolves #193

### CODE REVIEW

File: src/lib/action.ts [hunk 1]
```diff
@@ -237,10 +238,10 @@ More details printed to console.`
       return "invalid latex";
     }
 
-    const img = await this.canvas.addImage(
+    const img: FabricTeXImage = await this.canvas.addImage(
```
Comment (pihart) [inline src/lib/action.ts:241]: This isn't really necessary (we don't care about how much stronger this type is than `Image`) but it conveys intent

File: src/lib/page.ts [hunk 2]
```diff
@@ -103,16 +104,28 @@ export default class Page extends fabric.Canvas {
...
+   */
+  addImage = async <T extends fabric.IImageOptions>(
     imageURL: string,
     cursor?: Partial<Cursor>,
-    options?: fabric.IImageOptions
-  ): Promise<fabric.Image> =>
-    new Promise<fabric.Image>((resolve) =>
+    options?: T
+  ): Promise<fabric.Image & (typeof options extends undefined ? unknown : T)> =>
+    new Promise((resolve) =>
```
Comment (pihart) [inline src/lib/page.ts:119]: We are now allowed to remove the redundant type from the `new Promise()` call!

File: src/lib/pages.ts [hunk 3]
```diff
@@ -42,6 +42,7 @@ export default class Pages {
   savePage = (): void => {
     this.pagesJSON[this.currentIndex] = this.canvas.toObject([
       "id",
```
Comment (pihart) [inline src/lib/pages.ts:44]: This should probably be abolished in the future; store in the `data` property instead

**Review by cjquines — APPROVED** [review]

**Comment by cjquines** [comment]
i've begun to change my mind and am tentatively approving this

(sorry for not merging this before #189, but am still thinking about this)

**Comment by pihart** [comment]
That's okay, things should be good now (conflict was due to rebasing #189 on merge)

**Pull request merged**